### ISSUE TITLE
Value assignment converts text to date

### ISSUE THREAD

**Issue by StevenFunte** [ClosedXML/ClosedXML#1869]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [x ] Yes
- [ ] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**
0.96.0 from NUGet

**What is the current behavior?**

When I assign the string"Z12.31" to a cell value it gets converted to a date type.
This is the only value that behaves like this.

This is what is displayed in the excel file.
A1 - 12/30/2022 19:00
A2  -  Z12_31

I have tested this with VS 2019 and VS 2022. .net Framework 4.7 and .net core 3.1 and6.0.

**What is the expected behavior or new feature?**

That it doesn't convert a string ("Z12.31") to a date.

**Is this a regression from the previous version?**

Regressions get higher priority. Test against the latest build of the previous minor version. For example, if you experience a problem on v0.95.3, check whether it the problem occurred in v0.94.2 too. 

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#

using Excel = ClosedXML.Excel;

namespace ConsoleApp1
{
    internal class Program
    {
        static void Main(string[] args)
        {
            Excel.XLWorkbook xlsWorkbook = new Excel.XLWorkbook();

            Excel.IXLWorksheet xlsWorksheet = xlsWorkbook.AddWorksheet("Test");

            xlsWorksheet.Cell("A1").DataType = Excel.XLDataType.Text;
            xlsWorksheet.Cell("A1").Value = "Z12.31";
            xlsWorksheet.Cell("A2").DataType = Excel.XLDataType.Text;
            xlsWorksheet.Cell("A2").Value = "Z12_31";


            string fileName = string.Format(@"C:\temp\Test_{0}.xlsx", DateTime.Now.ToString("yyyy_MM_dd_HH_mm"));
            xlsWorkbook.SaveAs(fileName);

        }
    }
}
[Test_2022_10_07_15_56.xlsx](https://github.com/ClosedXML/ClosedXML/files/9737121/Test_2022_10_07_15_56.xlsx)

- [ x] I attached a sample spreadsheet.  (You can drag files on to this issue)

**Comment by igitur** [comment]
That's because dotnet parses the string as a valid date. Related discussion at https://stackoverflow.com/questions/43339856/why-does-ad3ad08-represent-a-valid-date-in-the-net-framework

**Comment by igitur** [comment]
Also related:

https://github.com/ClosedXML/ClosedXML/wiki/Text-with-numbers-are-getting-converted-to-numbers

**Comment by jahav** [comment]
In this case, it is not an era, but a Z indicates a Zulu timezone, not an era. See [reference source](https://referencesource.microsoft.com/#mscorlib/system/globalization/datetimeparse.cs,3737). Basically, parsing result is a datetime with a timezone without a time (0:00 is used and then moved).

Anyway, the real question in this case is if it is a reasonable behavior. I would say definitely no, but no idea how to simple fix. I don't see any API to forbid timezone without a time. I don't want to add a NodaTime as dependency for this.

Sidenote: The XLCell.Value setter is a kitchen sink (it allows to soo many undocumented types, eg. XLRangeBase). It should probably be replaced by a method like RudegoldbergValueGuessing and the .Value should be replaced by a setter with a clear semantic. /sigh, backward compatibility.

**Comment by jahav** [comment]
@igitur How about we limit parsing options only to formats the culture defines as a datetime? If I, as a human, have trouble parsing the date, ClosedXML shouldn't either. Usage of a library should be intuitive. 

```csharp
string input = "2021-10-14";
var workbookCulture = CultureInfo.InvariantCulture;
if (DateTime.TryParseExact(input, workbookCulture.DateTimeFormat.GetAllDateTimePatterns(), workbookCulture, DateTimeStyles.AllowLeadingWhite | DateTimeStyles.AllowTrailingWhite, out var result))
{
	Console.WriteLine(result);
}
```

The GetAllDateTimePatterns method has basically all sensible patterns for the culture. We should also add an explicitely specified workbook culture, as discussed in #1407. This is an example of formats for invariant culture
> MM/dd/yyyy 
yyyy-MM-dd 
dddd, dd MMMM yyyy 
dddd, dd MMMM yyyy HH:mm 
dddd, dd MMMM yyyy hh:mm tt 
dddd, dd MMMM yyyy H:mm 
dddd, dd MMMM yyyy h:mm tt 
dddd, dd MMMM yyyy HH:mm:ss 
MM/dd/yyyy HH:mm 
MM/dd/yyyy hh:mm tt 
MM/dd/yyyy H:mm 
MM/dd/yyyy h:mm tt 
yyyy-MM-dd HH:mm 
yyyy-MM-dd hh:mm tt 
yyyy-MM-dd H:mm 
yyyy-MM-dd h:mm tt 
MM/dd/yyyy HH:mm:ss 
yyyy-MM-dd HH:mm:ss 
MMMM dd 
MMMM dd 
yyyy'-'MM'-'dd'T'HH':'mm':'ss.fffffffK 
yyyy'-'MM'-'dd'T'HH':'mm':'ss.fffffffK 
ddd, dd MMM yyyy HH':'mm':'ss 'GMT' 
ddd, dd MMM yyyy HH':'mm':'ss 'GMT' 
yyyy'-'MM'-'dd'T'HH':'mm':'ss 
HH:mm 
hh:mm tt 
H:mm 
h:mm tt 
HH:mm:ss 
yyyy'-'MM'-'dd HH':'mm':'ss'Z' 
dddd, dd MMMM yyyy HH:mm:ss 
yyyy MMMM 
yyyy MMMM

**Comment by jahav** [comment]
This will be solved once we moved to type system #1894. After that, ClosedXML won't try to guess a type for a type conversion (unless setting cell value through a method that explicitly tries to do that).

Basically if you set the value to string, even `"2022-11-04"`, it will stay a string. If you want a date, use will `DateTime(2022, 11, 4)` (though date is mostly polite function, for all intents and purposes it is a number).
